tools/harness-thci/OpenThread.py

```diff
@@ -3003,6 +3220,214 @@ def setCSLsuspension(self, suspend):
+
+    def __getMlIid(self):
+        """get the Mesh Local IID."""
+        print('%s call __getMlIid' % self.port)
+        # getULA64() would return the full string representation
+        mleid = ModuleHelper.GetFullIpv6Address(self.getULA64()).lower()
+        mliid = mleid[-19:].replace(':', '')
+        print('mliid: %s' % mliid)
+        return mliid
+
+    def __setMlIid(self, sMlIid):
+        """Set the Mesh Local IID before Thread Starts."""
+        assert ':' not in sMlIid
+        cmd = 'mliid %s' % sMlIid
+        self.__executeCommand(cmd)
+
+    @API
+    def registerDUA(self, destAddr='', sMleId='', sAddr=''):
+        # TODO: implement this
+        """TODO: only used for explicitly registration (DEV-1916/DEV-1923) """
+
+        # TODO-P1: workaround for DEV-2007
+        # overwrite generated DUA
+        if sAddr != '' and destAddr == '' and sMleId == '':
+            self.__setDUA(sAddr)
+            return
+
+        # TODO-P1: to be removed for pseudo operation which would be replaced by send_udp
+        cmd = 'dua reg'
+
+        if sAddr != '':
+            cmd += ' addr %s' % sAddr
+        if sMleId != '':
+            # sMleId is colon seperated, here converts.
+            cmd += ' mliid %s' % sMleId.replace(':', '')
+        if destAddr != '':
+            fullIp = ModuleHelper.GetFullIpv6Address(destAddr).lower()
+            print('dest fullip %s' % fullIp)
+            if fullIp.startswith(self.meshLocalPrefix.lower()[0:19]) and fullIp.startswith('0000:00ff:fe00:', 20):
+                dest = fullIp.split(':')[-1]
+                cmd += ' dest %s' % dest
+            else:
+                print('dest should be rloc address')
+                return False
+        print(cmd)
+        return self.__executeCommand(cmd)[-1] == 'Done'
+
+    @API
+    def config_next_mlr_status_rsp(self, status_code):
+        cmd = 'bbr mgmt mlr response %d' % status_code
+        return self.__executeCommand(cmd)[-1] == 'Done'
+
+    @API
+    def setMLRtimeout(self, iMsecs):
+        """Setup BBR MLR Timeout to `iMsecs` seconds."""
+        self.__configBbrDataset(MlrTimeout=iMsecs)
+
+    @API
+    def stopListeningToAddr(self, sAddr):
+        print('%s call stopListeningToAddr' % self.port)
+
+        # convert to list for single element, for possible extension
+        # requirements.
+        if not isinstance(sAddr, list):
+            sAddr = [sAddr]
+
+        for addr in sAddr:
+            cmd = 'ipmaddr del ' + addr
+            try:
+                self.__executeCommand(cmd)
+            except CommandError as ex:
+                if ex.code == OT_ERROR_ALREADY:
+                    pass
+                else:
+                    raise
+
+        return True
+
+    @API
+    def registerMulticast(self, sAddr='ff04::1234:777a:1'):
+        """subscribe to the given ipv6 address (sAddr) in interface and send MLR.req OTA
+
+        note: workaround agreed before finial decision discussed in the DEV-1819.
```

Comment:
not sure is it still required given fake option via `send_udp()`?

Reply:
`registerMulticast` actually subscribes to the MA, not just send fake MLR.req.

Reply:
Yes, we can remove the note.